### DISCUSSION TITLE
Option to move all windows of an application at once

### DISCUSSION THREAD

**Discussion by QuantumBJump** [rxhanson/Rectangle#326] [Ideas]
I tend to have multiple firefox windows open at once, and it sometimes gets annoying if I want to move firefox to another monitor, because I have to move one window with cmd-opt-ctrl-arrow, then cmd-\` to the next window, then cmd-opt-ctrl-arrow again, rinse and repeat for however many firefox windows I have open. I wonder if it would be possible/useful to have a setting (possibly toggleable with a shortcut? or maybe chorded?) which performs the same action to all windows of the active application.

**Comment by bryson** [comment]
I have the same issue here, but with Xcode, VS Code, etc.

Typically I keep all of my windows for a single app on the same monitor anyway, so moving entire apps' collections of windows is as common for me as moving a single window. Especially because I use a laptop with an external monitor. When unplugging the monitor, everything moves back to the laptop. When plugging it back in, the windows never go back to the places they were before.

After having written this comment, I realize that every time I move them from monitor to monitor after plugging back in my external, I also need to maximize all of my VS Code windows the same, for example. Perhaps a more generalized solution could just be adding a modifier that changes a command from "the current window" to "all the current app's windows"? 

Either way, this feature would be very useful for me and I'm sure others.

**Comment by rxhanson** [comment]
A while back, I did end up adding this into my paid superset of Rectangle, [Rectangle Pro](https://rectangleapp.com/pro). The primary reason for not adding it into Rectangle first is just to keep Rectangle as simple as possible. Having less actions in Rectangle makes it a little more approachable by keeping the UI from deviating too far from Spectacle & Magnet.

**Reply by abernier** [reply]
pressing a key to add the "all windows" behavior would be ideal, for ex: `<all-windows-key> ctrl+alt+enter` => maximize all windows
